packages/block-library/src/template-part/edit/placeholder/index.js

```diff
@@ -28,6 +33,20 @@ export default function TemplatePartPlaceholder( {
 	const { saveEntityRecord } = useDispatch( coreStore );
 	const [ step, setStep ] = useState( PLACEHOLDER_STEPS.initial );
 
+	const { areaIcon, areaLabel } = useSelect( ( select ) => {
+		const definedAreas = select(
+			editorStore
+		).__experimentalGetDefaultTemplatePartAreas();
+
+		const selectedArea = find( definedAreas, { area } );
+		const defaultArea = find( definedAreas, { area: 'uncategorized' } );
+
+		return {
+			areaIcon: selectedArea?.icon || defaultArea?.icon,
+			areaLabel: selectedArea?.label || __( 'Template Part' ),
+		};
+	}, [] );
```

Comment:
My bad for the confusing comments before!

The empty dependency was if we kept the previous approach, which didn't use any dependencies.
In this case, we are using `area`, which should be listed as dependency.

TBH I don't think it would make a big difference anyway?
Once the block is inserted I don't think there's a way to change the variation (and therefore the `area` attribute). 🤔

Reply:
All correct.  It shouldn't make a difference as we can't change the area attribute on a placeholder, however we should add it for consistency.